x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elastic/authorization/ElasticInferenceServiceAuthorizationHandler.java

```diff
@@ -9,129 +9,313 @@
+                        waitTime,
+                        serviceComponents.threadPool().executor(UTILITY_THREAD_POOL_NAME)
+                    )
+            );
+        } catch (Exception e) {
+            logger.warn("Failed scheduling authorization request", e);
+        }
+    }
+
+    private void scheduleAndSendAuthorizationRequest() {
+        if (shutdown.get()) {
+            return;
+        }
+
+        scheduleAuthorizationRequest();
+        sendAuthorizationRequest();
+    }
+
+    private void sendAuthorizationRequest() {
+        try {
+            ActionListener<ElasticInferenceServiceAuthorizationModel> listener = ActionListener.wrap((model) -> {
+                setAuthorizedContent(model);
+                if (callback != null) {
+                    callback.run();
                 }
-                requestCompleteLatch.countDown();
+                firstAuthorizationCompletedLatch.countDown();
             }, e -> {
-                Throwable exception = e;
-                if (e instanceof ElasticsearchWrapperException wrapperException) {
-                    exception = wrapperException.getCause();
-                }
-
-                logger.warn(Strings.format(FAILED_TO_RETRIEVE_MESSAGE + " Encountered an exception: %s", exception));
-                listener.onResponse(ElasticInferenceServiceAuthorization.newDisabledService());
-                requestCompleteLatch.countDown();
+                // we don't need to do anything if there was a failure, everything is disabled by default
+                firstAuthorizationCompletedLatch.countDown();
             });
 
-            var request = new ElasticInferenceServiceAuthorizationRequest(baseUrl, getCurrentTraceInfo());
-
-            sender.sendWithoutQueuing(logger, request, AUTH_RESPONSE_HANDLER, DEFAULT_AUTH_TIMEOUT, newListener);
+            authorizationHandler.getAuthorization(listener, sender);
         } catch (Exception e) {
-            logger.warn(Strings.format("Retrieving the authorization information encountered an exception: %s", e));
-            requestCompleteLatch.countDown();
+            logger.warn("Failure while sending the request to retrieve authorization", e);
+            // we don't need to do anything if there was a failure, everything is disabled by default
+            firstAuthorizationCompletedLatch.countDown();
         }
     }
 
-    private TraceContext getCurrentTraceInfo() {
-        var traceParent = threadPool.getThreadContext().getHeader(Task.TRACE_PARENT);
-        var traceState = threadPool.getThreadContext().getHeader(Task.TRACE_STATE);
+    private synchronized void setAuthorizedContent(ElasticInferenceServiceAuthorizationModel auth) {
+        logger.debug("Received authorization response");
+        var authorizedTaskTypesAndModels = authorizedContent.get().taskTypesAndModels.merge(auth)
+            .newLimitedToTaskTypes(EnumSet.copyOf(implementedTaskTypes));
```

Comment:
As a performance improvement we could check if the new authorization response is the same as the previous on and skip the rest of the logic here.